src/options.ts

```diff
@@ -125,6 +272,81 @@ function addCountup(builder: PanelOptionsEditorBuilder<ClockOptions>) {
     });
 }
 
+//---------------------------------------------------------------------
+// DESCRIPTION
+//---------------------------------------------------------------------
+function addDescription(builder: PanelOptionsEditorBuilder<ClockOptions>) {
+  const category = ['Description'];
+
+  builder
+    .addRadio({
+      category,
+      path: 'descriptionSettings.source',
+      name: 'Source',
+      settings: {
+        options: [
+          { value: DescriptionSource.none, label: 'None' },
+          { value: DescriptionSource.input, label: 'Input' },
+        ],
+      },
+      defaultValue: DescriptionSource.none,
+      showIf: (o) => {
+        let show =
+          (o.mode === ClockMode.countup && o.countupSettings.source !== ClockSource.query) ||
+          (o.mode === ClockMode.countdown && o.countdownSettings.source !== ClockSource.query);
+
+        if (show && o.descriptionSettings.source === DescriptionSource.query) {
+          o.descriptionSettings.source = DescriptionSource.none;
+        }
+
+        return show;
+      },
+    })
+    .addRadio({
+      category,
+      path: 'descriptionSettings.source',
+      name: 'Source',
+      settings: {
+        options: [
+          { value: DescriptionSource.none, label: 'None' },
+          { value: DescriptionSource.input, label: 'Input' },
+          { value: DescriptionSource.query, label: 'Query' },
+        ],
+      },
+      defaultValue: DescriptionSource.none,
+      showIf: (o) =>
+        (o.mode === ClockMode.countup && o.countupSettings.source === ClockSource.query) ||
+        (o.mode === ClockMode.countdown && o.countdownSettings.source === ClockSource.query),
+    })
+    .addTextInput({
+      category,
+      path: 'descriptionSettings.description',
```

Comment:
To avoid this getting confusing let's not call this "description" you can maybe it called `descriptionText` or `descriptionContent` otherwise it is not clear what is inside this in the rest of the code.